### PR TITLE
Update Gaufrette packages to resolve deprecates in the old version 

### LINKED ISSUE #9074: MIME type and extension not being set for remote asset uploaded via API
## Description:
After uploading a remote asset via the Mautic API, the `mime` and `extension` fields for the asset are returned as null. However, performing the same upload via the UI sets these fields and you are able to preview the asset.

### Steps to reproduce:
1. Make POST request to `/assets/new` endpoint:

   ```bash
   curl --request POST 'https://example.com/api/assets/new' \
   --data-urlencode 'file=http://assetserver.com/asset.pdf' \
   --data-urlencode 'storageLocation=remote' \
   --data-urlencode 'title=asset.pdf'
   ```
2. The `extension` and `mime` fields in the response body are null.

### PR DESCRIPTION
**Pull request by mabumusa1**
| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | 3.1
| Bug fix?                               | yes
| New feature?                           | no
| Deprecations?                          | yes
| BC breaks?                             | no
| Automated tests included?              | no
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #9074... 

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Configure the AWS S3 Plugin
3. Create a remote asset
4. Access the remote asset through the API

### CODE REVIEW

**Comment by mabumusa1** [comment]
@StevenChin please test this PR on your server and let me know if you find any errors

**Comment by StevenChin** [comment]
@mabumusa1 thanks for submitting a PR, I will review and test later today.

**Comment by RCheesley** [comment]
@cla-bot check

**Comment by RCheesley** [comment]
@mabumusa1 think there might be something wrong with your git configuration judging by the author on the commit.

**Comment by StevenChin** [comment]
@mabumusa1 I did some testing on my end but I still wasn't able to see the results I was expecting. Let me know if I'm doing something wrong here:

**Steps to reproduce**

1. Make request to upload remote file via API:

   ```bash
    curl --request POST 'https://mautibox.com/9120/api/assets/new' \
    --header 'Content-Type: application/x-www-form-urlencoded' \
    --data-urlencode 'file=http://africau.edu/images/default/sample.pdf' \
    --data-urlencode 'storageLocation=remote' \
    --data-urlencode 'title=test-asset.pdf'
   ```

2. Observe that both `asset.mime` and `asset.extension` in the response body are null:

Postman screenshot

**Comment by mabumusa1** [comment]
@StevenChin Good catch, here is the solution for it 

image

**Comment by StevenChin** [comment]
@mabumusa1 thanks for the fix! I was able to test and I can confirm that it is working as expected now.

I'm a bit curious about the solution. Do you think that those calls to set the entity were just missed by the original implementors?

**Comment by RCheesley** [comment]
@mabumusa1 looks like this needs a bit of attention to the test coverage to bring it up to 'does not change' or 'improves' coverage.

@dennisameling I'd like to get a code review from you in addition to the 👍  from @StevenChin before we go for merging this.

**Comment by mabumusa1** [comment]
@RCheesley this is almost done, please review it

**Comment by RCheesley** [comment]
Bumping to 3.1.2 to give more time for testing.

**Comment by mabumusa1** [comment]
> @mabumusa1 thanks for the fix! I was able to test and I can confirm that it is working as expected now.
> 
> I'm a bit curious about the solution. Do you think that those calls to set the entity were just missed by the original implementors?

@StevenChin yes that is correct, I was wondering if you can write a review for this PR, it will help us get it merged for the next release

File: app/bundles/AssetBundle/Tests/Controller/Api/AssetApiControllerFunctionalTest.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            'storageLocation' => 'remote',
+            'title'           => 'title',
+        ];
+        $this->client->request('POST', 'api/assets/new', $payload);
+        $clientResponse = $this->client->getResponse();
+        $response       = json_decode($clientResponse->getContent(), true);
+        $this->assertEquals($payload['title'], $response['asset']['title']);
+        $this->assertEquals($payload['storageLocation'], $response['asset']['storageLocation']);
+        $this->assertStringContainsString('application/pdf', $response['asset']['mime']);
+        $this->assertNotNull($response['asset']['extension']);
```
Comment (StevenChin) [inline app/bundles/AssetBundle/Tests/Controller/Api/AssetApiControllerFunctionalTest.php]: Should the extension be `pdf` here?

File: app/bundles/AssetBundle/Tests/Controller/Api/AssetApiControllerFunctionalTest.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            'title'           => 'title',
+        ];
+        $this->client->request('POST', 'api/assets/new', $payload);
+        $clientResponse = $this->client->getResponse();
+        $response       = json_decode($clientResponse->getContent(), true);
+        $this->assertEquals($payload['title'], $response['asset']['title']);
+        $this->assertEquals($payload['storageLocation'], $response['asset']['storageLocation']);
+        $this->assertStringContainsString('text/plain', $response['asset']['mime']);
+        $this->assertNotNull($response['asset']['size']);
+        $this->assertNotNull($response['asset']['extension']);
```
Comment (StevenChin) [inline app/bundles/AssetBundle/Tests/Controller/Api/AssetApiControllerFunctionalTest.php]: Similarly, should the extension be txt here?

File: app/bundles/AssetBundle/Controller/Api/AssetApiController.php [hunk 3]
```diff
@@ -70,6 +70,8 @@ protected function prepareParametersForBinding($parameters, $entity, $action)
                 }
             } elseif ('remote' === $parameters['storageLocation']) {
                 $parameters['remotePath'] = $parameters['file'];
+                $entity->setStorageLocation('remote');
```
Comment (StevenChin) [inline app/bundles/AssetBundle/Controller/Api/AssetApiController.php:74]: After taking another look at this, would it make sense to use same logical flow as [AssetController->newAction()](https://github.com/mautic/mautic/blob/staging/app/bundles/AssetBundle/Controller/AssetController.php#L281) i.e:

```php
// populate entity from request parameters
$entity->setTitle($parameters['title']);
$entity->setStorageLocation('remote');
$entity->setRemotePath($parameters['remotePath']);

// upload entity
$entity->preUpload();
$entity->upload();
```

**Comment by StevenChin** [comment]
> > @mabumusa1 thanks for the fix! I was able to test and I can confirm that it is working as expected now.
> > I'm a bit curious about the solution. Do you think that those calls to set the entity were just missed by the original implementors?
> 
> @StevenChin yes that is correct, I was wondering if you can write a review for this PR, it will help us get it merged for the next release

Thanks for the info and sure no problem, I submitted a review.

**Comment by npracht** [comment]
@mabumusa1 can you check last review then we can merge it into 3.1.2 :)

File: app/bundles/AssetBundle/Tests/Controller/Api/AssetApiControllerFunctionalTest.php (hunk 1 above)
Comment (mabumusa1) [reply app/bundles/AssetBundle/Tests/Controller/Api/AssetApiControllerFunctionalTest.php]: yes, because it is a test case on a pdf file

Comment (StevenChin) [reply app/bundles/AssetBundle/Tests/Controller/Api/AssetApiControllerFunctionalTest.php]: Indeed, so we can update the assertion to be more specific.

**Comment by mabumusa1** [comment]
> @mabumusa1 can you check last review then we can merge it into 3.1.2 :)

I think this is ready for merge

**Comment by npracht** [comment]
@cla-bot check

**Comment by RCheesley** [comment]
@mabumusa1 it looks like a small conflict to be addressed, may require a rebase for Codecov against staging also.

**Pull request merged**